main/src/main/scala/sbt/internal/CommandExchange.scala

```diff
@@ -167,21 +168,35 @@ private[sbt] final class CommandExchange {
   }
 
   def publishEvent[A: JsonFormat](event: A): Unit = {
+    val broadcastStringMessage = true
     val toDel: ListBuffer[CommandChannel] = ListBuffer.empty
     event match {
       case entry: StringEvent =>
+        val params = toLogMessageParams(entry)
         channels.foreach {
           case c: ConsoleChannel =>
-            if (entry.channelName.isEmpty || entry.channelName == Some(c.name)) {
+            if (broadcastStringMessage) {
```

Comment:
Was this for debug purposes? Is `broadcastStringMessage` meant to be wired up to something, rather than be a constant true?

Reply:
I think the way it was before it correct: it should take into account channel name, otherwise you things get messed up..

Reply:
This was done on purpose, so on terminal client you still see the compiler warning etc coming from the VS Code client etc.

Reply:
> compiler warning etc coming from the VS Code client